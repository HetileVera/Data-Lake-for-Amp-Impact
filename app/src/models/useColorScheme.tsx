import { useTheme } from "@chakra-ui/react";

export function useColorScheme() {
  const colorScheme = useTheme().name;
  const tone = (value: number | undefined) => (value ? `${colorScheme}.${value}` : colorScheme);

  return { tone, colorScheme };
}
