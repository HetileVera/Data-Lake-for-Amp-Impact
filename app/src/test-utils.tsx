import React from "react";
import { render, RenderOptions } from "@testing-library/react";
import { ChakraProvider } from "@chakra-ui/react";

import { theme } from "themes/orange";

export const CustomChakraProvider = ({ children }: { children: React.ReactNode }) => {
  return (
    <ChakraProvider theme={theme}>
      <div data-testid="test-root">{children}</div>
    </ChakraProvider>
  );
};

const customRender = (ui: React.ReactElement, options?: Omit<RenderOptions, "wrapper">) => render(ui, { wrapper: CustomChakraProvider, ...options });

export * from "@testing-library/react";

export { customRender as render };
