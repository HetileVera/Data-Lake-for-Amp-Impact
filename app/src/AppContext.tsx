import React, { createContext, useContext } from "react";
import isEmpty from "lodash/isEmpty";
import throttle from "lodash/throttle";

import { AppStore, IAppStore } from "./models/AppStore";
import { IInstallation } from "./models/Installation";
import { isLocalDev, websiteUrlOrigin, assetWebsiteUrl, secretKey } from "./helpers/settings";
import { onSnapshot } from "mobx-state-tree";

// Create an instance of the root store (a.k.a appStore)
const appStore = AppStore.create({});

// Store state to local storage
const storeLocally = throttle(() => appStore.saveLocal(), 5000);

// Store state in S3 when applicable
const storeRemotely = throttle(() => appStore.saveRemote(), 30000);

// The initialization function for the root store
export function initializeStore() {
  if (!isEmpty(secretKey)) console.warn("Access key is bundled!");

  if (isLocalDev) {
    console.log(`isLocalDev '${isLocalDev}'`);
    console.log(`websiteUrlOrigin '${websiteUrlOrigin}'`);
    console.log(`assetWebsiteUrl '${assetWebsiteUrl}'`);

    if (window.location.host !== "localhost:3000") {
      console.warn("You should run the app on localhost!");

      throw new Error("App should run on localhost for development mode!");
    }
  }

  appStore.loadLocal();
  onSnapshot(appStore, (snapshot) => {
    storeLocally();
    storeRemotely();
  });
}

// Create the store context
export const StoreContext = createContext<IAppStore | undefined>(undefined);

// Create the store provider component
export const StoreProvider: React.FunctionComponent = ({ children }) => {
  return <StoreContext.Provider value={appStore}>{children}</StoreContext.Provider>;
};

// useStore hook
export function useStore(): IAppStore {
  const store = useContext(StoreContext);

  if (!store) {
    throw new Error("useStore() must be used within a StoreProvider");
  }

  return store;
}

// useInstallation hook
export function useInstallation(): IInstallation {
  const store = useStore();
  const installation = store.installation;

  if (!installation) {
    throw new Error("An installation instance was not created yet");
  }

  return installation;
}
