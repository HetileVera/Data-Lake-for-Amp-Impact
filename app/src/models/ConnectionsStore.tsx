import { useEffect } from "react";
import isEmpty from "lodash/isEmpty";
import { types, Instance, getParent } from "mobx-state-tree";

import { useInstallation } from "AppContext";
import { listAppflowConnections } from "api/list-appflow-connections";

import { BaseStore, isStoreNew, isStoreError, isStoreReady, isStoreLoading, isStoreReLoading } from "./BaseStore";
import { ICredentials } from "./helpers/Credentials";
import { IInstallation } from "./Installation";

/**
 * Represents the store for the AppFlow Salesforce connection names
 */
export const ConnectionsStore = BaseStore.named("ConnectionsStore")
  .props({
    connectionNames: types.array(types.string),
  })

  .views((self) => ({
    get empty() {
      return isEmpty(self.connectionNames);
    },

    get credentials(): ICredentials {
      const parent: IInstallation = getParent(self);
      return parent.credentials;
    },

    get region(): string {
      const parent: IInstallation = getParent(self);
      return parent.region;
    },
  }))

  .actions((self) => {
    const superReset = self.reset;
    return {
      reset() {
        superReset();
        self.connectionNames.clear();
      },

      async doLoad() {
        const accessKey = self.credentials.accessKey;
        const secretKey = self.credentials.secretKey;
        const region = self.region;

        const rawConnectionNames = await listAppflowConnections({ accessKey, secretKey, region });
        self.runInAction(() => {
          self.connectionNames.replace(rawConnectionNames);
        });
      },
    };
  });

// see https://mobx-state-tree.js.org/tips/typescript
export interface IConnectionsStore extends Instance<typeof ConnectionsStore> {}

export function useConnectionsStore() {
  const installation = useInstallation();
  const store = installation.connectionsStore;

  useEffect(() => {
    if (!isStoreNew(store)) return;
    store.load();
  }, [store]);

  return {
    isError: isStoreError(store),
    isReady: isStoreReady(store),
    isLoading: isStoreLoading(store),
    isReloading: isStoreReLoading(store),
    store,
  };
}
