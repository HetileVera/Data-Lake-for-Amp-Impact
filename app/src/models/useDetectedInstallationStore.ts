import { isStoreError, isStoreLoading, isStoreReady, isStoreReLoading } from "models/BaseStore";

import { DetectedInstallationStore } from "./DetectedInstallationsStore";

// Create an instance of this store
const store = DetectedInstallationStore.create({});

export function useDetectedInstallationStore() {
  return {
    isError: isStoreError(store),
    isReady: isStoreReady(store),
    isLoading: isStoreLoading(store),
    isReloading: isStoreReLoading(store),
    store,
  };
}
