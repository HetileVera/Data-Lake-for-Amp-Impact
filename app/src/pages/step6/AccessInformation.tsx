import React from "react";
import { Box, Heading } from "@chakra-ui/react";
import { observer } from "mobx-react";

import { useColorScheme } from "models/useColorScheme";

import { AccessInfoGrid } from "./AccessInfoGrid";

export const AccessInformation = observer(() => {
  const { tone } = useColorScheme();

  return (
    <>
      <Heading id="step6-h2-access-information" size="md" pt="0px" pb="30px" color={tone(600)} letterSpacing="-1px">
        Access Information
      </Heading>

      <Box display="block" fontSize="sm" mb={6}>
        You will connect your data lake to Tableau using Amazon Athena, a natively supported connector for Tableau Desktop and Tableau Cloud. This
        information will be used inside of Tableau when making the connection.
      </Box>

      <AccessInfoGrid />
    </>
  );
});
