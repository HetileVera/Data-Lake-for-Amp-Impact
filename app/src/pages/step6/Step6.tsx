import React from "react";
import { useNavigate } from "react-router-dom";
import { Box, Button, Container, Heading, HStack, Text, useTheme } from "@chakra-ui/react";
import { ArrowBackIcon } from "@chakra-ui/icons";
import { observer } from "mobx-react";

import { useInstallation } from "AppContext";
import { Header } from "components/Header";
import { CurvedBox } from "components/CurvedBox";
import { StepsBanner } from "components/StepsBanner";
import { AccessInformation } from "pages/step6/AccessInformation";
import { UsersPanel } from "pages/step6/UsersPanel";
import { useColorScheme } from "models/useColorScheme";
import { LaunchYourConnection } from "./LauncYourConnection";
import { AdminTools } from "./AdminTools";

export const Step6 = observer(() => {
  const theme = useTheme();
  const { tone, colorScheme } = useColorScheme();
  const installation = useInstallation();
  const navigate = useNavigate();
  const step = installation.createUsersStep;

  React.useEffect(() => {
    step.markStarted();
    window.scrollTo(0, 0);
  }, [step]);

  const handlePrevious = () => {
    navigate("/steps/5");
  };

  return (
    <>
      <CurvedBox bgGradient={theme.gradients.bgLight} />

      <Header />

      <Box position="relative">
        <CurvedBox />
        <StepsBanner current={6} />

        <Container maxW="container.md" pt="16px" position="relative" pb={10}>
          <Box float="right" fontSize="small" textAlign="center" color={tone(100)}>
            <Text>Installation ID</Text>
            <Text>{installation.id}</Text>
          </Box>

          <Heading display="inline-block" size="lg" pt="0px" pb="10px" color="white" letterSpacing="-1px">
            Connect to Analytics
          </Heading>

          <Box color={tone(50)}>
            Now that you have the data lake provisioned, you can connect to Tableau for analytics. Here you will gather all the necessary information
            needed to make that connection.
          </Box>

          <Box borderRadius="lg" boxShadow="base" bg={tone(25)} mt={4} p={10} pb={4} position="relative">
            <UsersPanel />
          </Box>

          <Box borderRadius="lg" boxShadow="base" bg={tone(25)} mt={4} p={10} pb={4} position="relative">
            <AccessInformation />
          </Box>

          <Box borderRadius="lg" boxShadow="base" bg={tone(25)} mt={4} p={10} pb={4} position="relative">
            <LaunchYourConnection />
          </Box>

          <Box borderRadius="lg" boxShadow="base" bg={tone(25)} mt={4} p={10} pb={4} position="relative">
            <AdminTools />
          </Box>

          <HStack justifyContent="space-between" p={3} pt={6} mb={12}>
            <Button
              colorScheme={colorScheme}
              size="md"
              _hover={{ bg: tone(100) }}
              leftIcon={<ArrowBackIcon />}
              variant="ghost"
              onClick={handlePrevious}
            >
              Previous
            </Button>
          </HStack>
        </Container>
      </Box>
    </>
  );
});
