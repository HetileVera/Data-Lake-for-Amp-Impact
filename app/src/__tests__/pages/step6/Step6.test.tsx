import { cleanup, screen } from "@testing-library/react";
import { create } from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";

import { render, RenderWithChakra } from "test-utils";

import * as appContext from "AppContext";
import * as metadataStore from "models/MetadataStore";
import * as importStatusStore from "models/ImportStatusStore";
import { Step6 } from "pages/step6/Step6";

const spyScrollTo = jest.fn();

describe("Step6 component", () => {
  beforeEach(() => {
    Object.defineProperty(global.window, "scrollTo", { value: spyScrollTo });
  });

  afterEach(() => {
    spyScrollTo.mockClear();
    cleanup;
  });

  test("snapshot", () => {
    // @ts-ignore
    jest.spyOn(appContext, "useInstallation").mockImplementation(() => {
      return {
        id: "1",
        startDate: "2022-01-01",
        startedBy: "test-user",
        accountId: "test-account-id",
        region: "us-east-1",
        appFlowConnectionName: "test-appflow-connection-name",
        credentials: [],
        nextStepNumber: "2",
        createUsersStep: {
          markStarted: jest.fn(),
          markCompleted: jest.fn(),
        },
      };
    });
    // @ts-ignore
    jest.spyOn(metadataStore, "useMetadataStore").mockImplementation(() => {
      return {
        isReady: true,
        store: {
          selectedObjects: [{}],
        },
      };
    });
    // @ts-ignore
    jest.spyOn(importStatusStore, "useImportStatusStore").mockImplementation(() => {
      return {
        isError: false,
        store: {
          loadingError: false,
          selectedObjects: [{}],
        },
      };
    });

    const tree = create(
      <RenderWithChakra>
        <appContext.StoreProvider>
          <BrowserRouter>
            <Step6 />
          </BrowserRouter>
        </appContext.StoreProvider>
      </RenderWithChakra>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test("render properly", () => {
    // @ts-ignore
    jest.spyOn(appContext, "useInstallation").mockImplementation(() => {
      return {
        id: "1",
        startDate: "2022-01-01",
        startedBy: "test-user",
        accountId: "test-account-id",
        region: "us-east-1",
        appFlowConnectionName: "test-appflow-connection-name",
        credentials: [],
        nextStepNumber: "2",
        deploymentOperations: {
          isInProgress: true,
          isSuccess: true,
          operations: [],
        },
        createUsersStep: {
          markStarted: jest.fn(),
          markCompleted: jest.fn(),
        },
        triggerDeployment: jest.fn(),
      };
    });
    // @ts-ignore
    jest.spyOn(metadataStore, "useMetadataStore").mockImplementation(() => {
      return {
        isReady: true,
        store: {
          selectedObjects: [{}],
        },
      };
    });
    // @ts-ignore
    jest.spyOn(importStatusStore, "useImportStatusStore").mockImplementation(() => {
      return {
        isError: false,
        store: {
          loadingError: false,
          selectedObjects: [{}],
        },
      };
    });

    render(
      <appContext.StoreProvider>
        <BrowserRouter>
          <Step6 />
        </BrowserRouter>
      </appContext.StoreProvider>
    );

    expect(screen.getAllByText(/connect to analytics/i)).toBeTruthy();
    expect(screen.getAllByText(/region/i)).toBeTruthy();
    expect(screen.getByText(/now that you have the data lake provisioned/i)).toBeInTheDocument();
    expect(screen.getByText("Previous")).toBeInTheDocument();
    expect(screen.queryByText("Next")).not.toBeInTheDocument();
  });
});
