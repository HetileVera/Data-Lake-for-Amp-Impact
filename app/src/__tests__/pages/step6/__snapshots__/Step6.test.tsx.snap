// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step6 component snapshot 1`] = `
Array [
  <div
    data-testid="test-root"
  >
    <div
      className="css-0"
      id="step6"
    >
      <div
        className="css-1v55i00"
        data-testid="curved-box"
      />
      <div
        className="chakra-container css-1szj4qk"
      >
        <div
          className="chakra-stack css-olw0lh"
        >
          <div
            className="chakra-stack css-18lyn6q"
          >
            <div
              className="css-1i3fafl"
            >
              <img
                className="chakra-image css-0"
                src="Salesforce_Corporate_Logo_RGB.png"
              />
            </div>
            <div
              className="chakra-stack__divider css-r2vkku"
            />
            <div
              className="css-nutmff"
            >
              <img
                className="chakra-image css-ir5tdx"
                src="AWS_logo_RGB_BLK.png"
              />
            </div>
          </div>
          <div
            className="chakra-stack css-1n7gsjr"
          >
            <div
              className="css-18n4g2t"
            >
              <h2
                className="chakra-heading css-1opsxio"
                id="h2-title"
              >
                Data Lake for Nonprofits, Powered by AWS
              </h2>
            </div>
          </div>
        </div>
      </div>
      <div
        className="css-79elbk"
      >
        <div
          className="css-t7j6xp"
          data-testid="curved-box"
        />
        <div
          className="css-4e82ru"
        >
          <div
            className="chakra-container css-h66qep"
          >
            <div
              className="chakra-stack css-84zodg"
            >
              <span
                className="css-9xsoz7"
              >
                1
              </span>
              <span
                className="css-9xsoz7"
              >
                2
              </span>
              <span
                className="css-9xsoz7"
              >
                3
              </span>
              <span
                className="css-9xsoz7"
              >
                4
              </span>
              <span
                className="css-9xsoz7"
              >
                5
              </span>
              <span
                className="css-hfqsm"
              >
                6
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        className="chakra-container css-11yz6e2"
      >
        <h2
          className="chakra-heading css-noy8un"
        >
          Connect to Analytics
        </h2>
        <div
          className="css-1kveix6"
        >
          <p
            className="chakra-text css-0"
          >
            Region: 
            us-east-1
          </p>
          <p
            className="chakra-text css-0"
          >
            ID: 
            1
          </p>
        </div>
        <div
          className="css-vfbt1y"
        >
          <div
            className="chakra-stack css-1s6236i"
          >
            <div
              className="css-1y0bycm"
            >
              <div
                className="css-0"
              >
                Now that you have the data lake provisioned, you can connect to Tableau for analytics. Here you will gather all the necessary information needed to make that connection.
              </div>
            </div>
          </div>
        </div>
        <div
          className="css-1jkeem6"
        >
          <div
            className="chakra-stack css-1uiig4k"
          >
            <h2
              className="chakra-heading css-lx7khp"
            >
              Users
            </h2>
            <button
              aria-label="Check again"
              className="chakra-button css-xroa0q"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              <svg
                aria-hidden={true}
                className="chakra-icon css-onkibi"
                focusable={false}
                viewBox="0 0 24 24"
              >
                <g
                  fill="currentColor"
                >
                  <path
                    d="M10.319,4.936a7.239,7.239,0,0,1,7.1,2.252,1.25,1.25,0,1,0,1.872-1.657A9.737,9.737,0,0,0,9.743,2.5,10.269,10.269,0,0,0,2.378,9.61a.249.249,0,0,1-.271.178l-1.033-.13A.491.491,0,0,0,.6,9.877a.5.5,0,0,0-.019.526l2.476,4.342a.5.5,0,0,0,.373.248.43.43,0,0,0,.062,0,.5.5,0,0,0,.359-.152l3.477-3.593a.5.5,0,0,0-.3-.844L5.15,10.172a.25.25,0,0,1-.2-.333A7.7,7.7,0,0,1,10.319,4.936Z"
                  />
                  <path
                    d="M23.406,14.1a.5.5,0,0,0,.015-.526l-2.5-4.329A.5.5,0,0,0,20.546,9a.489.489,0,0,0-.421.151l-3.456,3.614a.5.5,0,0,0,.3.842l1.848.221a.249.249,0,0,1,.183.117.253.253,0,0,1,.023.216,7.688,7.688,0,0,1-5.369,4.9,7.243,7.243,0,0,1-7.1-2.253,1.25,1.25,0,1,0-1.872,1.656,9.74,9.74,0,0,0,9.549,3.03,10.261,10.261,0,0,0,7.369-7.12.251.251,0,0,1,.27-.179l1.058.127a.422.422,0,0,0,.06,0A.5.5,0,0,0,23.406,14.1Z"
                  />
                </g>
              </svg>
            </button>
          </div>
          <div
            className="css-1y54b5v"
          >
            Create users who will need access to the data. These users will have 
            <i>
              least privilege
            </i>
             credentials to access only the data and resources they need.
          </div>
          <div
            className="css-1l0ccn2"
          >
            <p
              className="chakra-text css-7jk7y8"
            >
              When using Tableau to connect to your data lake, you need AWS IAM access keys. Access keys are associated with users. Add a user to obtain the access keys.
            </p>
            <div
              className="css-s2uf1z"
            >
              <button
                className="chakra-button css-406sl6"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                Add User
              </button>
            </div>
          </div>
        </div>
        <div
          className="css-1jkeem6"
        >
          <h2
            className="chakra-heading css-1081m9x"
            id="step6-h2-access-information"
          >
            Access Information
          </h2>
          <div
            className="css-bsg1we"
          >
            You will connect your data lake to Tableau using Amazon Athena, a natively supported connector for Tableau Desktop and Tableau Cloud. This information will be used inside of Tableau when making the connection.
          </div>
          <div
            className="css-1eaeoz1"
          >
            <div
              className="css-msgkkf"
            >
              Server
            </div>
            <div
              className="css-f6z7d2"
            >
              <div
                className="chakra-stack css-12dr6av"
              >
                <div
                  className="css-ffath9"
                />
                <svg
                  className="chakra-icon css-w4ub0m"
                  data-testid="copy-icon"
                  focusable={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseDown={[Function]}
                  onMouseEnter={[Function]}
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm3 4H8c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h11c1.1 0 2-.9 2-2V7c0-1.1-.9-2-2-2zm0 16H8V7h11v14z"
                    fill="currentColor"
                  />
                </svg>
              </div>
            </div>
            <div
              className="css-msgkkf"
            >
              Port
            </div>
            <div
              className="css-f6z7d2"
            >
              <div
                className="chakra-stack css-12dr6av"
              >
                <div
                  className="css-ffath9"
                />
                <svg
                  className="chakra-icon css-w4ub0m"
                  data-testid="copy-icon"
                  focusable={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseDown={[Function]}
                  onMouseEnter={[Function]}
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm3 4H8c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h11c1.1 0 2-.9 2-2V7c0-1.1-.9-2-2-2zm0 16H8V7h11v14z"
                    fill="currentColor"
                  />
                </svg>
              </div>
            </div>
            <div
              className="css-msgkkf"
            >
              S3 Staging Directory
            </div>
            <div
              className="css-f6z7d2"
            >
              <div
                className="chakra-stack css-12dr6av"
              >
                <div
                  className="css-ffath9"
                />
                <svg
                  className="chakra-icon css-w4ub0m"
                  data-testid="copy-icon"
                  focusable={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseDown={[Function]}
                  onMouseEnter={[Function]}
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm3 4H8c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h11c1.1 0 2-.9 2-2V7c0-1.1-.9-2-2-2zm0 16H8V7h11v14z"
                    fill="currentColor"
                  />
                </svg>
              </div>
            </div>
            <div
              className="css-msgkkf"
            >
              Catalog
            </div>
            <div
              className="css-f6z7d2"
            >
              <div
                className="chakra-stack css-12dr6av"
              >
                <div
                  className="css-ffath9"
                />
                <svg
                  className="chakra-icon css-w4ub0m"
                  data-testid="copy-icon"
                  focusable={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseDown={[Function]}
                  onMouseEnter={[Function]}
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm3 4H8c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h11c1.1 0 2-.9 2-2V7c0-1.1-.9-2-2-2zm0 16H8V7h11v14z"
                    fill="currentColor"
                  />
                </svg>
              </div>
            </div>
            <div
              className="css-msgkkf"
            >
              Workgroup
            </div>
            <div
              className="css-f6z7d2"
            >
              <div
                className="chakra-stack css-12dr6av"
              >
                <div
                  className="css-ffath9"
                />
                <svg
                  className="chakra-icon css-w4ub0m"
                  data-testid="copy-icon"
                  focusable={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseDown={[Function]}
                  onMouseEnter={[Function]}
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm3 4H8c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h11c1.1 0 2-.9 2-2V7c0-1.1-.9-2-2-2zm0 16H8V7h11v14z"
                    fill="currentColor"
                  />
                </svg>
              </div>
            </div>
            <div
              className="css-msgkkf"
            >
              Database
            </div>
            <div
              className="css-f6z7d2"
            >
              <div
                className="chakra-stack css-12dr6av"
              >
                <div
                  className="css-ffath9"
                >
                  public
                </div>
                <svg
                  className="chakra-icon css-w4ub0m"
                  data-testid="copy-icon"
                  focusable={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseDown={[Function]}
                  onMouseEnter={[Function]}
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm3 4H8c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h11c1.1 0 2-.9 2-2V7c0-1.1-.9-2-2-2zm0 16H8V7h11v14z"
                    fill="currentColor"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          className="css-1jkeem6"
        >
          <h2
            className="chakra-heading css-1081m9x"
          >
            Launch Your Connection in Tableau
          </h2>
          <div
            className="css-bsg1we"
          >
            Launch your Tableau application.
             
            <a
              className="chakra-link css-157cndf"
              href="https://help.tableau.com/current/pro/desktop/en-us/examples_amazonathena.htm"
              rel="noopener noreferrer"
              target="_blank"
            >
              Utilize the User and Access Information from above to complete the steps outlined in this help document 
              <svg
                className="chakra-icon css-onkibi"
                focusable={false}
                viewBox="0 0 24 24"
              >
                <g
                  fill="none"
                  stroke="currentColor"
                  strokeLinecap="round"
                  strokeWidth="2"
                >
                  <path
                    d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"
                  />
                  <path
                    d="M15 3h6v6"
                  />
                  <path
                    d="M10 14L21 3"
                  />
                </g>
              </svg>
            </a>
             
            inside of Tableau Desktop or Tableau Cloud.
          </div>
          <h2
            className="chakra-heading css-2c3uw2"
            onClick={[Function]}
          >
            Special Considerations for using Tableau Desktop
            <svg
              className="chakra-icon css-xbxoa7"
              focusable={false}
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                fill="currentColor"
              />
            </svg>
          </h2>
          <h2
            className="chakra-heading css-5ig38"
            onClick={[Function]}
          >
            If you are using Tableau Cloud you will need to...
            <svg
              className="chakra-icon css-xbxoa7"
              focusable={false}
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                fill="currentColor"
              />
            </svg>
          </h2>
        </div>
        <div
          className="css-1jkeem6"
        >
          <h2
            className="chakra-heading css-1mrywam"
            onClick={[Function]}
          >
            Admin Tools
            <svg
              className="chakra-icon css-xbxoa7"
              focusable={false}
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                fill="currentColor"
              />
            </svg>
          </h2>
        </div>
        <div
          className="chakra-stack css-1aoc1cb"
        >
          <button
            className="chakra-button css-a63yqb"
            onClick={[Function]}
            type="button"
          >
            <span
              className="chakra-button__icon css-1wh2kri"
            >
              <svg
                aria-hidden={true}
                className="chakra-icon css-onkibi"
                focusable={false}
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
                  fill="currentColor"
                />
              </svg>
            </span>
            Previous
          </button>
        </div>
      </div>
    </div>
  </div>,
  <span />,
]
`;
